lib/modules/manager/crossplane/extract.spec.ts

```diff
@@ -0,0 +1,155 @@
+import { codeBlock } from 'common-tags';
+import { Fixtures } from '../../../../test/fixtures';
+import { extractPackageFile } from '.';
+
+const validPackages = Fixtures.get('validPackages.yml');
+const malformedPackages = Fixtures.get('malformedPackages.yml');
+const randomManifest = Fixtures.get('randomManifest.yml');
+const mixedManifest = Fixtures.get('mixedManifest.yml');
+
+describe('modules/manager/crossplane/extract', () => {
+  describe('extractPackageFile()', () => {
+    it('returns null for empty', () => {
+      expect(extractPackageFile('nothing here', 'packages.yml')).toBeNull();
+    });
+
+    it('returns null for invalid', () => {
+      expect(
+        extractPackageFile(`${malformedPackages}\n123`, 'packages.yml'),
+      ).toBeNull();
+    });
+
+    it('return null for kubernetes manifest', () => {
+      const result = extractPackageFile(randomManifest, 'packages.yml');
+      expect(result).toBeNull();
+    });
+
+    it('return invalid-value if deps are not valid images and ignore if missing', () => {
```

Comment:
This test is doing two things at once. Would it make sense to split it into two parts, something like this?

- return invalid-value if deps are not valid images
- if deps are missing, ignore them